src/mui/input/SelectManyInput.js

```diff
@@ -0,0 +1,167 @@
+ * <SelectManyInput source="gender" choices={choices} />
+ *
+ * You might want to pass the type='object', instead of 'text'. If then, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <SelectManyInput source="gender" type="object" choices={choices} />
+ *
+ * The `type="embedded"` will result the array of choices with full json object inside.
+ * If you just want the result value is array of ids, please use `type="object"`.
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <SelectManyInput source="gender" type="embedded" choices={choices} />
+ *
+ * Values => ['M']
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <SelectManyInput source="author_id" type="embedded" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * The object passed as `options` props is passed to the material-ui-chip-input component
+ * @see https://github.com/TeamWertarbyte/material-ui-chip-input
+ */
+export class SelectManyInput extends Component {
+    // Comment input onBlur because SelectManyInput don't pass correct values on this event
+    handleBlur = (eventOrValue) => {
+        // this.props.onBlur(eventOrValue);
+        // this.props.input.onBlur(eventOrValue);
+    };
+    // Comment input onFocus because SelectManyInput don't pass correct values on this event
+    handleFocus = (event) => {
+        // this.props.onFocus(event);
+        // this.props.input.onFocus(event);
+    };
+
+    handleChange = (eventOrValue) => {
+        if (this.props.type == 'object') {
+            eventOrValue = this.extractIds(eventOrValue);
+        }
+        this.props.onChange(eventOrValue);
+        this.props.input.onChange(eventOrValue);
+    };
+
+    extractIds = (eventOrValue) => {
+        let value = eventOrValue;
+        if (value.target && value.target.value) {
+            value = value.target.value;
+        }
+        if (Array.isArray(value)) {
+            return value.map((o) => o[this.props.optionValue])
+        }
+        return value;
+    };
+
+    filterObjectByIds = (ids) => {
+        return this.props.choices.filter((o) => ids.indexOf(o[this.props.optionValue]) >= 0);
+    };
+
+    render() {
+        const {
+            elStyle,
+            input,
+            choices,
+            label,
+            meta: { touched, error },
+            options,
+            optionText,
+            optionValue,
+            resource,
+            source,
+            type,
+        } = this.props;
+
+        // Always use uncontrolled mode
```

Comment:
I don't understand. On the contrary, we should use controlled mode, shouldn't we?